### ISSUE TITLE
Simulator won't run app unless manifest is named "manifest.webapp"

### ISSUE THREAD

**Issue by havwig** [mozilla/r2d2b2g#695]
For my first app, I named the manifest file "appname.manifest" and installed it to the simulator. The app shows up with the name from the manifest in the simulator dashboard, and I get no error messages. Seems like a successful install.

However, I get no icon in the simulator, and no response when I press "Run" in the dashboard. Not even a warning or an error in the console log. After a bit of experimentation I renamed the manifest file to "manifest.webapp", and then installation on simulated device and running from dashboard worked as expected.

I'm not sure if the name actually _has_ to be "manifest.webapp", but in that case I would expect a warning in the simulator dashboard. The following two files reproduces the problem in Simulator 3.0.1 / Firefox 22.

index.html:

``` html
<!doctype html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Hello FxOS</title>
  </head>
  <body>
    <h1>Hello!</h1>
  </body>
</html>
```

hello.manifest:

``` json
{
  "name": "Hello",
  "description": "Simplest possible app",
  "launch_path": "/index.html"
}
```

**Comment by nickdesaulniers** [comment]
Yes I think we're checking literally for manifest.webapp when we should just be checking for a file ending with the .webapp extension.  I don't think the manifest has to be named manifest.webapp in b2g, but I'll double check when I have a free minute.  Thanks @havwig for reporting this!

**Comment by mykmelez** [comment]
For a hosted app, the manifest does not need to be named manifest.webapp, and the Simulator accepts any name for the manifest, f.e.: http://areatweet.com/app.manifest

For a packaged app, however, the manifest _does_ need to be named manifest.webapp, per: https://developer.mozilla.org/en-US/docs/Web/Apps/Packaged_apps#Differences_from_hosted_apps

And it looks like you're developing a packaged app. Therefore the Simulator is correctly failing to install and run it. However, it certainly is an problem that the Simulator didn't explain why it couldn't install/run your app. It should clearly explain the issue and how to fix it!

**Comment by mcepl** [comment]
I see another problem here: when I **HAVE** manifest called manifest.webapp for _HOSTED_ app, simulator decides that my app is packaged, although it isn't. Is it a known bug?

**Comment by nickdesaulniers** [comment]
@mcepl , you must be clicking "Add Directory" and then browsing to a local copy of the manifest.  To install an app as hosted, you must type in the URL of the manifest.webapp file then click "Add Manifest" ("Add URL" changes to "Add Manifest" if it validates the manifest).  Otherwise you can make an app out of any page by typing in a URL of any page that's not a valid manifest file and the simulator will create an app for you.

**Comment by mcepl** [comment]
@nickdesaulniers  could I add file:/// URL, e.g., file:///home/matej/projekty/hesla/hesla.webapp ?
